src/lib.js

```diff
@@ -41,6 +43,44 @@ var curlConverter = {
         return str;
     },
 
+    addQueryParamsFromDataOption: function(curlObj, urlData, request) {
+        // What this will do:
+        // If URL is http://example.com?a=b and -d 'c=d' => http://example.com?a=b&c=d
+        // If URL is http://example.com#fragment and -d 'c=d' => http://example.com#fragment
+        if ((curlObj.uploadFile.length > 0 || curlObj.head || curlObj.data.length > 0) && curlObj.get) {
+            if (request.url.includes('?')) {
+                request.url += '&' + urlData;
+            }
+            else {
+                request.url += '?' + urlData;
+            }
+        }
+    },
+
+    getRequestMethod: function(curlObj) {
+        // RFC- https://curl.haxx.se/docs/manpage.html
+        // checking if the user has mentioned -T or --upload-file in curl command
+        if (curlObj.uploadFile.length > 0) {
+            return "PUT";
+        }
+        // checking if the user has mentioned -I or --head in curl command
+        else if (curlObj.head) {
+            return "HEAD";
+        }
+        // checking if the user has mentioned -G or --get in curl command
+        else if (curlObj.get) {
+            return "GET";
+        }
+        // checking if the user has mentioned any of these (-d, --data, --data-binary, --data-ascii) in curl command
+        else if (curlObj.data.length > 0 || curlObj.dataAscii.length > 0 || curlObj.dataUrlencode.length > 0 || curlObj.dataBinary) {
+            return "POST";
+        }
+        // set method to GET if no param is present
+        else {
```

Comment:
no need of an else block here. you can directly return "GET" as the default case.